java/ql/src/utils/model-generator/ModelGeneratorUtils.qll

```diff
@@ -151,11 +151,29 @@ predicate isRelevantTaintStep(DataFlow::Node node1, DataFlow::Node node2) {
   )
 }
 
+predicate isRelevantContent(DataFlow::Content f) {
+  isRelevantType(f.(DataFlow::FieldContent).getField().getType()) or
+  f instanceof DataFlow::ArrayContent or
+  f instanceof DataFlow::CollectionContent or
+  f instanceof DataFlow::MapKeyContent or
+  f instanceof DataFlow::MapValueContent
+}
+
+string parameterNodeAsInput(DataFlow::ParameterNode p) {
+  result = parameterAccess(p.asParameter())
+  or
+  result = "Argument[-1]" and p instanceof DataFlow::InstanceParameterNode
+}
+
 string returnNodeAsOutput(TargetAPI api, ReturnNodeExt node) {
   if node.getKind() instanceof ValueReturnKind
```

Comment:
Isn't `api` unbound in this branch of the `if` statement?

Reply:
Indeed it is. We're probably being saved by magic at the moment. But yeah, that ought to be fixed.

Reply:
From the looks of it, there is nothing that explicitly ties the returnnode to the TargetApi in captureSource (I am unfamiliar with the code, so this might be wrong). Will try and push a change that makes this explicit as also removes the TargetApi from the returnNodeAsOutput predicate.